method_comparison/text_generation_benchmark/run.py

```diff
@@ -0,0 +1,389 @@
+# Copyright 2025-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Main entry point to run the experiments. Contains general setup and the proper inference code.
+"""
+
+import argparse
+import gc
+import json
+import os
+import sys
+import time
+from typing import Optional
+
+import bitsandbytes
+import torch
+import transformers
+from data import prepare_benchmark_prompts
+from transformers import AutoModelForCausalLM, AutoTokenizer, BitsAndBytesConfig, set_seed
+from utils import (
+    BenchmarkConfig,
+    BenchmarkResult,
+    BenchmarkStatus,
+    get_memory_usage,
+    init_cuda,
+    log_results,
+    validate_experiment_path,
+)
+
+import peft
+from peft import PeftConfig, get_peft_model
+
+
+def load_base_results(model_id: str) -> Optional[dict]:
+    """Load base model results if they exist."""
+    base_results_dir = os.path.join(os.path.dirname(__file__), "base_results")
+    model_name = model_id.replace("/", "_").replace("-", "_")
+    filename = f"base_{model_name}.json"
+    filepath = os.path.join(base_results_dir, filename)
+
+    if os.path.exists(filepath):
+        with open(filepath) as f:
+            return json.load(f)
+    return None
+
+
+def measure_inference_time(
+    model, tokenizer, prompts, max_new_tokens, num_runs, print_fn, category_generation_params=None
+):
+    """Measure inference time for each prompt category."""
+    inference_times = {}
+    time_per_token = {}
+    generated_tokens = {}
+    individual_samples = {}
+
+    for category, category_prompts in prompts.items():
+        print_fn(f"\nMeasuring inference time for {category} prompts...")
+        category_times = []
+        category_tokens = []
+        category_time_per_token = []
+        category_samples = []
+
+        for prompt in category_prompts:
+            prompt_times = []
+            prompt_tokens = []
+            prompt_time_per_token = []
+
+            # Prepare input
+            inputs = tokenizer(prompt, return_tensors="pt").to(model.device)
+
+            # Get category-specific max_new_tokens
+            cat_max_new_tokens = (category_generation_params.get(category, {}).get("max_new_tokens", max_new_tokens))
```

Comment:
> how this could be none ?

In `run_base.py`, `category_generation_params` are not passed, so the default value of `None` is used. Try running `python run_base.py` and you should encounter the error. Let's remove the default value for `category_generation_params` and instead require the dict to always be passed.